advice-generator.js

```diff
@@ -0,0 +1,41 @@
+import { allCharacters, randomCharacter, randomQuote } from "./characters.js";
+
+function zip(list1, list2) {
+  return list1.map((element, i) => {
+    return [element, list2[i]];
+  });
+}
+
+export function selectCharacterQuotes(userInput) {
+  let charactersSelected = [];
+  let quotesSelected = [];
+  // If any of the words in the user's input match a quote title by any character, select that quote
+  if (userInput != undefined) {
+    allCharacters.forEach((character) => {
+      const allQuotes = Object.keys(character.quotes);
+      allQuotes.forEach((quote) => {
+        if (userInput.toLowerCase().includes(quote)) {
+          charactersSelected.push(character.name);
+          quotesSelected.push(character.quotes[quote]);
+        }
+      });
+    });
+  }
+
+  //If there are fewer than 3 quotes selected, choose quotes randomly until three are selected
+  while (quotesSelected.length < 3) {
+    let characterToAdd = randomCharacter();
+    let characterName = characterToAdd.name;
+    // Make sure that only 1 quote from each character is chosen
+    while (charactersSelected.includes(characterName)) {
+      characterToAdd = randomCharacter();
+      characterName = characterToAdd.name;
+    }
+    charactersSelected.push(characterName);
+    // Choose a random quote from that character
+    quotesSelected.push(randomQuote(characterToAdd));
+  }
+
+  const charactersAndQuotes = zip(charactersSelected, quotesSelected);
```

Comment:
You could remove the need for `zip` entirely. Since you're always consecutively `push`ing to both `charactersSelected` and `quotesSelected` only to later `zip` them together, I'd change `charactersAndQuotes` to be an object where each property is a character name whose value is the quote. Then, you could get the object's `keys` for an array of character names when determining if a character's been added yet.